### PR TITLE
add configuration option to enable compatibility zone whilst running devMode

### PR DESCRIPTION
**Pull request by roastario**

### CODE REVIEW

**Comment by tlil** [comment]
Seems like you've hit a flaky test as well :(

**Comment by mikehearn** [comment]
Yeah, I'd say there could be some docs for this new config option?

**Comment by tlil** [comment]
@roastario can you add info to https://docs.corda.net/head/corda-configuration-file.html and https://docs.corda.net/head/changelog.html ?

**Comment by roastario** [comment]
docs added.

File: node/src/main/kotlin/net/corda/node/services/config/NodeConfiguration.kt [hunk 1]
```diff
@@ -256,16 +256,19 @@ data class NodeConfigurationImpl(
     private fun validateDevModeOptions(): List<String> {
         if (devMode) {
             compatibilityZoneURL?.let {
-                return listOf("'compatibilityZoneURL': present. Property cannot be set when 'devMode' is true.")
+                if (devModeOptions?.allowCompatibilityZone != true) {
+                    return listOf("'compatibilityZoneURL': present. Property cannot be set when 'devMode' is true.")
```
Comment (tlil) [inline node/src/main/kotlin/net/corda/node/services/config/NodeConfiguration.kt]: This message is now a bit misleading, no?

File: docs/source/corda-configuration-file.rst [hunk 2]
```diff
@@ -183,7 +183,13 @@ absolute path to the node's base directory.
     :doormanURL: Root address of the network registration service.
     :networkMapURL: Root address of the network map service.
 
-.. note:: Only one of ``compatibilityZoneURL`` or ``networkServices`` should be used.
+        .. note:: Only one of ``compatibilityZoneURL`` or ``networkServices`` should be used.
+
+:devModeOptions: Allows modification of certain devMode features
```
Comment (tlil) [inline docs/source/corda-configuration-file.rst]: devMode => ``devMode`` ?

**Comment by tlil** [comment]
And can you please update the changelog too :) Thanks!

File: docs/source/corda-configuration-file.rst [hunk 3]
```diff
@@ -325,3 +331,7 @@ Example adding/overriding keyStore password when starting Corda node:
 .. sourcecode:: shell
 
     java -Dcorda.rpcSettings.ssl.keyStorePassword=mypassword -jar node.jar
+
+
+Securing sensitive
```
Comment (tlil) [inline docs/source/corda-configuration-file.rst]: What's this?

Comment (roastario) [reply docs/source/corda-configuration-file.rst]: got over excited.

File: docs/source/changelog.rst [hunk 4]
```diff
@@ -147,7 +149,7 @@ Version 3.1
 * Update the fast-classpath-scanner dependent library version from 2.0.21 to 2.12.3
 
   .. note:: Whilst this is not the latest version of this library, that being 2.18.1 at time of writing, versions
-     later than 2.12.3 (including 2.12.4) exhibit a different issue.
+later than 2.12.3 (including 2.12.4) exhibit a different issue.
```
Comment (tlil) [inline docs/source/changelog.rst]: Breaks formatting

File: docs/source/changelog.rst [hunk 5]
```diff
@@ -965,15 +967,15 @@ Special thank you to `Qian Hong <https://github.com/fracting>`_, `Marek Skocovsk
 to Corda in M10.
 
 .. warning:: Due to incompatibility between older version of IntelliJ and gradle 3.4, you will need to upgrade Intellij
-   to 2017.1 (with kotlin-plugin v1.1.1) in order to run Corda demos in IntelliJ. You can download the latest IntelliJ
+to 2017.1 (with kotlin-plugin v1.1.1) in order to run Corda demos in IntelliJ. You can download the latest IntelliJ
```
Comment (tlil) [inline docs/source/changelog.rst]: Breaks formatting

File: docs/source/changelog.rst [hunk 6]
```diff
@@ -965,15 +967,15 @@ Special thank you to `Qian Hong <https://github.com/fracting>`_, `Marek Skocovsk
...
    from `JetBrains <https://www.jetbrains.com/idea/download/>`_.
 
 .. warning:: The Kapt-generated models are no longer included in our codebase. If you experience ``unresolved references``
-   errors when building in IntelliJ, please rebuild the schema model by running ``gradlew kaptKotlin`` in Windows or
+errors when building in IntelliJ, please rebuild the schema model by running ``gradlew kaptKotlin`` in Windows or
```
Comment (tlil) [inline docs/source/changelog.rst]: Breaks formatting

File: docs/source/changelog.rst [hunk 7]
```diff
@@ -965,15 +967,15 @@ Special thank you to `Qian Hong <https://github.com/fracting>`_, `Marek Skocovsk
...
    ``./gradlew kaptKotlin`` in other systems. Alternatively, perform a full gradle build or install.
 
 .. note:: Kapt is used to generate schema model and entity code (from annotations in the codebase) using the Kotlin Annotation
-   processor.
+processor.
```
Comment (tlil) [inline docs/source/changelog.rst]: Breaks formatting

**Comment by tlil** [comment]
I've actually fixed it here, but haven't got push perms to your private repo mate

**Pull request merged**